File: backend/graph/state.js
// Shared state schema for the LangGraph pipeline
// All nodes read from and write to this state object

export const initialState = {
    userQuery: "",
    duration: 3,        // minutes (2-5)
    isValid: false,
    rejectionReason: "",
    content: "",        // Generated markdown walkthrough
    imagePrompt: "",
    imageUrl: "",
    audioBase64: "",
    error: null,
};

File: python-service/graph/state.py
from typing import TypedDict, Optional


class TutorState(TypedDict):
    user_query: str
    duration: int             # 2–5 minutes
    is_valid: bool
    rejection_reason: str
    content: str              # Generated markdown walkthrough
    system_prompt: str        # System prompt for LLM
    user_message: str         # User message for LLM
    error: Optional[str]
